opentelemetry-sdk/src/logs/log_processor.rs

```diff
@@ -299,55 +300,36 @@ pub struct BatchConfig {
 
 impl Default for BatchConfig {
     fn default() -> Self {
-        let mut config = BatchConfig {
+        BatchConfigBuilder::default().build()
+    }
+}
+
+/// A builder for creating [`BatchConfig`] instances.
+#[derive(Debug)]
+pub struct BatchConfigBuilder {
+    max_queue_size: usize,
+    scheduled_delay: Duration,
+    max_export_batch_size: usize,
+    max_export_timeout: Duration,
+}
+
+impl Default for BatchConfigBuilder {
+    /// Create a new [`BatchConfigBuilder`] initialized with default batch config values as per the specs.
+    /// The values are overriden by environment variables if set.
+    /// For a list of supported environment variables see [Batch LogRecord Processor](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/configuration/sdk-environment-variables.md#batch-logrecord-processor).
```

Comment:
Lets document it in this repo itself. Linking to spec is okay, but if spec adds something new and is not implemented here, it'll lead to confusion.